src/context/AuthContext/index.tsx

```diff
@@ -67,6 +67,18 @@ const AuthProvider = ({
   const [isLoading, setIsLoading] = useState(false);
   const [error, setError] = useState(false);
 
+  // handle signer null case when reloading window after clearing storage
+  const [refetchCounter, setRefetchCounter] = useState(0);
+  useEffect(() => {
+    if (signer || refetchCounter > 10) return;
+    const timeout = setInterval(async () => {
+      refetchSigner();
+      setRefetchCounter((counter) => counter + 1);
+    }, 500);
+
+    return () => clearInterval(timeout);
+  }, [signer]);
```

Comment:
spaces. Also no need to refetch if already defined no?
```suggestion
  const [refetchCounter, setRefetchCounter] = useState(0);
  
  useEffect(() => {
    if (signer || refetchCounter > 10) return;
    
    const timeout = setInterval(async () => {
    if(!signer) {
       refetchSigner();
       setRefetchCounter((counter) => counter + 1);
    }
    }, 500);

    return () => clearInterval(timeout);
  }, [signer]);
```

Reply:
@nivo33 If it's defined it won't refetch 

    if (signer || refetchCounter > 10) return;